addon/components/component-proxy.js

```diff
@@ -19,6 +19,27 @@ import Ember from 'ember'
 export default Component.extend({
   tagName: '',
 
+  /**
+   * Dynamically-generated layout to render the proxied component.
+   * @private
+   */
+  layout: computed('componentName', 'propsString', function() {
+    let { componentName, propsString } = this.getProperties('componentName', 'propsString');
+
+    // Disable linting for this line because the recommended way of importing
+    // results in a "Could not find module `@ember/template-compilation`" error ¯\_(ツ)_/¯
+    // eslint-disable-next-line ember/new-module-imports
+    return Ember.HTMLBars.compile(`
+      {{#if hasBlock}}
+        {{#component "${componentName}"${propsString}}}
+          {{yield}}
+        {{/component}}
+      {{else}}
+        {{component "${componentName}"${propsString}}}
+      {{/if}}
+    `);
+  }).readOnly(),
```

Comment:
How did this end up above `propNames`?

Reply:
because `layout` is a default prop of the component class that comes up on top along with `tagName`, `classNames` & [other attributes](https://github.com/smile-io/eslint-plugin-smile-ember/blob/master/lib/utils/property-order.js#L8)

Reply:
lol... but it's also a CP 😅 🤷‍♂️